common4j/src/test/com/microsoft/identity/common/java/platform/DeviceTest.java

```diff
@@ -0,0 +1,94 @@
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.common.java.platform;
+
+import com.microsoft.identity.common.java.AuthenticationConstants;
+import com.microsoft.identity.common.java.logging.DiagnosticContext;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+
+import lombok.val;
+
+/**
+ * Tests for {@link com.microsoft.identity.common.java.platform.Device}.
+ */
+public class DeviceTest {
+
+    @After
+    public void setUp() {
+        Device.clearDeviceMetadata();
+    }
+
+    @Test
+    public void testGetDataWhenMetadataIsNotSet(){
+        // Shouldn't crash.
+        val platformParameter = Device.getPlatformIdParameters();
+        Assert.assertEquals(3, platformParameter.size());
+        Assert.assertEquals("NOT_SET", platformParameter.get(Device.PlatformIdParameters.CPU_PLATFORM));
+        Assert.assertEquals("NOT_SET", platformParameter.get(Device.PlatformIdParameters.DEVICE_MODEL));
+        Assert.assertEquals("NOT_SET", platformParameter.get(Device.PlatformIdParameters.OS));
+
+        Assert.assertEquals("NOT_SET", Device.getManufacturer());
+        Assert.assertEquals("NOT_SET", Device.getModel());
+        Assert.assertEquals("NOT_SET", Device.getProductVersion());
+    }
+
+    @Test
+    public void testGetPlatformIdParameters(){
+        Device.setDeviceMetadata(new MockDeviceMetadata());
+
+        val platformParameter = Device.getPlatformIdParameters();
+        Assert.assertEquals(3, platformParameter.size());
+        Assert.assertEquals(MockDeviceMetadata.testCPU, platformParameter.get(Device.PlatformIdParameters.CPU_PLATFORM));
+        Assert.assertEquals(MockDeviceMetadata.testDeviceModel, platformParameter.get(Device.PlatformIdParameters.DEVICE_MODEL));
+        Assert.assertEquals(MockDeviceMetadata.testOS, platformParameter.get(Device.PlatformIdParameters.OS));
+    }
+
+    @Test
+    public void testGetProductVersion_DiagContextNotSet(){
+        Assert.assertEquals("NOT_SET", Device.getProductVersion());
+    }
+
+    @Test
+    public void testGetProductVersion(){
+        DiagnosticContext.INSTANCE.getRequestContext().put(AuthenticationConstants.SdkPlatformFields.VERSION, "TestVersion");
```

Comment:
Seems more like a test for DiagnosticContext than it is for DeviceMetadata

Reply:
We're still retrieving the value from DeviceMetadata. (Not saying it's my _preferred_ way, but I don't want to introduce unnecessary change for now).